src/Engine/GameAction.java

```diff
@@ -1106,4 +1110,128 @@ public XYCoord getTargetLocation()
       return destination;
     }
   } // ~UnitDeleteAction
+
+  public static class UnitProduceAction implements GameAction
+  {
+    final UnitProduce type;
```

Comment:
Should this (and Flare) be top-level `UnitActionType`s since they aren't CO-specific/custom actions?

Reply:
Flare, maybe, since we don't have a use-case yet for making Flares with different ranges/AoEs.
I don't want to have to make a new action type for every new unit type we want to have a unit build, though.